### DISCUSSION TITLE
Add Mermaid Chart Support

### DISCUSSION THREAD

**Discussion by leogdion** [SwiftPackageIndex/SwiftPackageIndex-Server#3346] [Ideas]
<a href="https://github.com/brightdigit/SublimationBonjour">
<figure>

<img width="895" alt="From https://github.com/brightdigit/SublimationBonjour" src="https://github.com/user-attachments/assets/a58ba2a3-56f3-4831-8e07-38569ad03445">
    <figcaption><em>from <strong>GitHub brightdigit/SublimationBonjour Repo</strong></em></figcaption>
</figure>
</a>

Mermaid is supported by Github Markdown...

<a href="https://swiftpackageindex.com/brightdigit/SublimationBonjour">
<figure>
<img width="902" alt="From " src="https://github.com/user-attachments/assets/7f1e4212-2e7f-4a7c-a34c-a5f8e460bea0"><figcaption>From Swift Package Index brightidigt/SublimationBonjour Page</figcaption>
</figure></a>

... but not the Swift Package Index site. 

It seems that [the javascript needs to be added to automatically parse these `pre` tags.](https://mermaid.js.org/intro/getting-started.html#_4-calling-the-mermaid-javascript-api)

I'd love to help add support to this if needed.

**Comment by heckj** [comment]
FWIW, it's also not supported in DocC - https://github.com/swiftlang/swift-docc-render/issues/413

**Comment by leogdion** [comment]
YES but the swift-package-index uses the README.md not the DocC so it's not necessarily out of ordinary.

**Comment by daveverwer** [comment]
What package is this from? I'll take a look.

**Reply by leogdion** [reply]
Apologies, I added links to the [brightdigit/SublimationBonjour repo](https://github.com/brightdigit/SublimationBonjour)

**Comment by daveverwer** [comment]
I took a look and I think this would be a good addition to the site if you wanted to work on it, Leo.

You’ll already find a `ReadmeController` in `readme_controller.js` and you’ll probably need to add a `connect` function to initialise the Mermaid rendering there. Looks like there’s a [standard package for rendering](https://www.npmjs.com/package/mermaid), so let’s use that. If you need to pre-process the HTML (which I don’t think will be necessary), we mainly do that in Swift in `PackageReadme+Model.swift`.

You can also find any readme-specific styles that you might need in `readme.scss`.

It should be a nicely self-contained bit of functionality. 👍

I’ll convert this into an issue now so we can track it and for any further discussion. I’m also happy to help out if you need any help getting up and running with the [front-end development environment](https://github.com/SwiftPackageIndex/SwiftPackageIndex-Server/blob/main/LOCAL_DEVELOPMENT_SETUP.md#developing-front-end-code) or with anything to do with the changes.

**Comment by daveverwer** [comment]
Tracked in #3348
